src/com/amazon/ion/impl/bin/IonManagedBinaryWriter.java

```diff
@@ -725,414 +124,279 @@ public boolean isFieldNameSet()
+            if(lst != tempLST) {//we found imported tables and created a new lst object so the references dont match.
+                flush();
+                lst = tempLST;
+                lstIndex = lst.getImportedMaxId();
+            }
+        }
     }
 
     public boolean isInStruct()
     {
-        return user.isInStruct();
+        return currentWriter.isInStruct();
     }
 
     // Write Value Methods
-
-    public void writeNull() throws IOException
-    {
-        user.writeNull();
+    public void writeNull() throws IOException {
+        currentWriter.writeNull();
     }
 
-    public void writeNull(final IonType type) throws IOException
-    {
-        user.writeNull(type);
+    public void writeNull(final IonType type) throws IOException {
+        currentWriter.writeNull(type);
     }
 
-    public void writeBool(final boolean value) throws IOException
-    {
-        user.writeBool(value);
+    public void writeBool(final boolean value) throws IOException {
+        currentWriter.writeBool(value);
     }
 
-    public void writeInt(long value) throws IOException
-    {
-        userState.writeInt(this, value);
-        user.writeInt(value);
+    public void writeInt(long value) throws IOException {
+        currentWriter.writeInt(value);
     }
 
-    public void writeInt(final BigInteger value) throws IOException
-    {
-        userState.writeInt(this, value);
-        user.writeInt(value);
+    public void writeInt(final BigInteger value) throws IOException {
+        currentWriter.writeInt(value);
     }
 
-    public void writeFloat(final double value) throws IOException
-    {
-        user.writeFloat(value);
+    public void writeFloat(final double value) throws IOException {
+        currentWriter.writeFloat(value);
     }
 
-    public void writeDecimal(final BigDecimal value) throws IOException
-    {
-        user.writeDecimal(value);
+    public void writeDecimal(final BigDecimal value) throws IOException {
+        currentWriter.writeDecimal(value);
     }
 
-    public void writeTimestamp(final Timestamp value) throws IOException
-    {
-        user.writeTimestamp(value);
+    public void writeTimestamp(final Timestamp value) throws IOException {
+        currentWriter.writeTimestamp(value);
     }
 
-    public void writeSymbol(String content) throws IOException
-    {
+    public void writeSymbol(String content) throws IOException {
         writeSymbolToken(intern(content));
     }
 
-    private boolean handleIVM(int sid) throws IOException {
-        if (user.isIVM(sid))
-        {
-            if (user.hasWrittenValuesSinceFinished())
-            {
-                // this explicitly translates SID 2 to an IVM and flushes out local symbol state
+    public void writeSymbolToken(SymbolToken token) throws IOException {
+        token = intern(token);
+        if (token != null && token.getSid() == ION_1_0_SID && user.getDepth() == 0 && !user.hasAnnotations()) {
```

Comment:
an IVM at the top level needs to trigger finish() all pending serialization needs to be completed

Reply:
SID 2 written by the user should not be interpreted as an IVM (see http://amzn.github.io/ion-docs/docs/symbols.html#ion-version-markers and http://amzn.github.io/ion-docs/guides/symbols-guide.html ). The only way for the user to trigger a new IVM to be written should be via finish.

I know this isn't technically related to LST append, but since you've already touched it here...